From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 12 Apr 2021 17:48:40 -0400
Subject: [PATCH] apps: Fix osmocom_fft on Windows, where /dev/null doesn't
 exist.

Replace "/dev/null" with os.devnull, which points to the correct place
on all platforms.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 apps/osmocom_fft | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/apps/osmocom_fft b/apps/osmocom_fft
index c48ac7e..1b96d28 100755
--- a/apps/osmocom_fft
+++ b/apps/osmocom_fft
@@ -29,6 +29,7 @@ from gnuradio.eng_option import eng_option
 from optparse import OptionParser
 from functools import partial
 
+import os
 import sys
 import signal
 import time
@@ -308,7 +309,7 @@ class app_top_block(gr.top_block, Qt.QMainWindow):
         except RuntimeError:
             self.freq = None
 
-        self.file_sink = blocks.file_sink(gr.sizeof_gr_complex, "/dev/null", False)
+        self.file_sink = blocks.file_sink(gr.sizeof_gr_complex, os.devnull, False)
         self.file_sink.set_unbuffered(False)
         self.file_sink.close() # close the sink immediately
         # lock/connect/unlock at record button event did not work, so we leave it connected at all times
-- 
2.34.1

